### DISCUSSION TITLE
correct  elastic hysteresis loop

### DISCUSSION THREAD

**Discussion by Ming-is** [neperfepx/FEPX#43] [Q&A] [answered]
I am trying to get the elastic hysteresis loop diagram with fepx. The parameter settings of the deformation history module in the config file are shown in the figure  below：
![图片1](https://user-images.githubusercontent.com/71123582/132283264-78b2aedb-8bfa-4851-86de-548e17a47c37.png)

But the curve I finally got is not quite correct. There is no negative strain in the figure：
![3_stress_strain_33](https://user-images.githubusercontent.com/71123582/132283759-3684d617-897f-4c59-97c7-14c09b9e1e63.png)

 Here are the simulation files and post-processing files I used. How can I get the correct elastic hysteresis loop? I hope you can help me solve my doubts：
[simulation.zip](https://github.com/acmelab-ua/FEPX/files/7118880/simulation.zip)

**Comment by mkasemer** [comment, marked as answer]
Hello @Ming-is, thanks for using FEPX.

I'm afraid I'm not sure what the issue is. The stress-strain curve that you show above looks proper to me. In the tensile portion of deformation, you begin to yield roughly around 300 (MPa?), and experience some plastic deformation. You similarly yield around -300 MPa in the compressive portion, again experiencing some amount of plasticity. When you begin to tensile deformation again, it looks as though you begin yielding at some level above the initial yield point, indicating correctly that there is some degree of hardening.

In other words, we should not necessarily expect that you will experience a negative strain by loading to the same magnitude in compression after tensile loading. Indeed, this is generally not the case if you experience plastic deformation in the initial tensile load: the plastic strain that is accrued during this initial load will make it more difficult (i.e., requiring more load) to experience a negative strain upon unloading/compressive loading. Consider another case: where the loading is fully elastic: in this scenario, tensile loading to a specific load/strain state should be similar to compressive loading to the same load/strain state (assuming no pronounced tension/compression asymmetry). The hysteresis loop you describe, upon introduction of plastic strain, is exactly what is limiting this behavior.

If you are attempting to hit a negative strain, you will have to increase the magnitude of the compressive load you are targeting.

**Reply by Ming-is** [reply]
it's  fine   ，thanks.
